physics/continuous_trajectory_body.hpp

```diff
@@ -442,32 +446,66 @@ ContinuousTrajectory<Frame>::ReadFromMessage(
             Polynomial<Position<Frame>, Instant>::template ReadFromMessage<
                 EstrinWithoutFMA>(polynomial));
       } else {
-        serialization::Polynomial const& polynomial = pair.polynomial();
-        if (polynomial.HasExtension(
-                serialization::PolynomialInMonomialBasis::extension) &&
+        not_null<serialization::Polynomial const*> polynomial =
```

Comment:
I find this pointer confusing as I have to wonder if it's going to dangle.  Let's go back to a reference and see my comment below.